src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/X509Pal.cs

```diff
@@ -31,15 +33,27 @@ private partial class AppleX509Pal : ManagedX509ExtensionProcessor, IX509Pal
                     switch (oid.Value)
                     {
                         case Oids.RsaRsa:
+                            if (key.IsInvalid)
+                            {
+                                throw Interop.AppleCrypto.CreateExceptionForOSStatus(errSecUnsupportedKeyFormat);
+                            }
                             return new RSAImplementation.RSASecurityTransforms(key);
                         case Oids.DsaDsa:
-                            if (key.IsInvalid) 
+                            if (key.IsInvalid)
                             {
                                 // SecCertificateCopyKey returns null for DSA, so fall back to manually building it.
                                 return DecodeDsaPublicKey(encodedKeyValue, encodedParameters);
                             } 
                             return new DSAImplementation.DSASecurityTransforms(key);
                         case Oids.Ecc:
+                            if (key.IsInvalid)
+                            {
+                                throw Interop.AppleCrypto.CreateExceptionForOSStatus(errSecUnsupportedKeyFormat);
```

Comment:
If this is just symmetry, please don't add it.

Reply:
Really my comment is "don't add uncover{able|ed} branches".  If you mean you expect an explicit parameters ECC key to end up here, I'd like to see a test that triggers it.

Alternatively, if named char-2 curve keys get here then it's a bad exception code, since the format was fine, but the ID / curve type was rejected (a different concern about adding this throw)

Reply:
It's a fair point that a test should be present for it. And if I can come up with some I will gladly add it. Then again, the failure was handled in the pre-Mojave code by the means of handling `SecCertificateCopyPublicKey` error code. Now it is not handled and the error spreads further through the code and causes unexpected failures much further from the actual failing code, which I find very counter-intuitive. I am merely restoring a behaviour that was present previously and that was already untested on older macOS versions.

Reply:
I guess I would be fine with putting `Debug.Assert` instead (and same for RSA above). Is that ok with you?

Reply:
Hmm, turns out the code path is actually hit with existing test:

```
     at Internal.Cryptography.Pal.X509Pal.AppleX509Pal.DecodePublicKey(Oid oid, Byte[] encodedKeyValue, Byte[] encodedParameters, ICertificatePal certificatePal) in /Users/filipnavara/Documents/GitHub/corefx/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/X509Pal.cs:line 46
     at Internal.Cryptography.Pal.CertificateExtensionsCommon.GetPublicKey[T](X509Certificate2 certificate, Predicate`1 matchesConstraints) in /Users/filipnavara/Documents/GitHub/corefx/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/CertificateExtensionsCommon.cs:line 34
     at System.Security.Cryptography.X509Certificates.ECDsaCertificateExtensions.GetECDsaPublicKey(X509Certificate2 certificate) in /Users/filipnavara/Documents/GitHub/corefx/src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/ECDsaCertificateExtensions.cs:line 22
     at System.Security.Cryptography.X509Certificates.Tests.PublicKeyTests.TestECDsaPublicKey_BrainpoolP160r1_ValidatesSignature(Byte[] curveData, Byte[] existingSignature) in /Users/filipnavara/Documents/GitHub/corefx/src/System.Security.Cryptography.X509Certificates/tests/PublicKeyTests.cs:line 403
```

Reply:
Looks like the second input-set into that Theory is, in fact, an explicit parameters public key.

I don't think it's failing for "brainpool", but "explicit parameters".

Reply:
Ok, will update the comment to reflect that.

Reply:
Hmm, checked it and it fails for both of the curves in the test data.

Reply:
Looks like the old `SecCertificateCopyPublicKey` API returns `errSecInvalidKeyRef` in all the error cases. Would that be generic enough?

Reply:
`errSecInvalidKeyRef` sounds fine / nice and compatible.